examples/accelerate/simple_cv_example.ipynb

```diff
@@ -0,0 +1,479 @@
+{
```

Comment:
<p>Maybe mention that it's because CUDA can't be initialized twice (once for single-GPU in the ntoebook and once for multi-GPUs in the training function). That's what PyTorch's error should also indicate, so this way seeing this error message will tell the user what to look for.</p><p> </p><p>Also maybe add a comment that it's fine to debug in the notebook and have calls to cuda, but the final training launch will require cleanup and a restart?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/196/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
<p>I went with something like so:</p><p> </p><p>In the intro:</p><blockquote>The following cell will restart Jupyter after writing the configuration, as CUDA code was called to perform this. CUDA can't be initialized more than once (once for the single-GPU's notebooks use by default, and then what would be again when <code>notebook_launcher</code> is called). It's fine to debug in the notebook and have calls to CUDA, but remember that in order to finally train a full cleanup and restart will need to be performed, such as what is shown below:</blockquote><p> </p><p>In <code>Preparing the Dataset and Model</code>:</p><p> </p><blockquote>Next you should prepare your dataset. As mentioned at earlier, great care should be taken when preparing the <code>DataLoaders</code> and model to make sure that **nothing** is put on *any* GPU. </blockquote><blockquote>If you do, it is recommended to put that specific code into a function and call that from within the notebook launcher interface, which will be shown later. </blockquote><p> </p>